python/dataset.cpp

```diff
@@ -269,7 +286,17 @@ void init_dataset(py::module &m) {
       m, "DataArrayView", R"(
         View for DataArray, representing a sliced view onto a DataArray, or an item of a Dataset;
         Mostly equivalent to DataArray, see there for details.)");
-  dataArrayView.def(py::init<DataArray &>());
+
+  options.disable_function_signatures();
+  dataArrayView.def(py::init<DataArray &>(), py::arg("dataArray"),
+                    R"(__init__(self, dataArray: DataArray) -> None
+                    
+                    DataArrayView initialiser.
+
+                    :param dataArray: Viewed DataArray.
+                    :type dataArray: DataArray
+                    )");
```

Comment:
I would actually prefer finding a way of removing this from the docs. Users should not have to use this. For Python users views should behave just like the actual thing, and docs should not distinguish between them.

Reply:
Do we want to remove the documentation for DataArrayView in this PR or should we keep it around until other references have been cleaned up? Removing it if we wat to is a one liner.

Reply:
Not sure, probbly keep for now?